watertap/property_models/activated_sludge/asm1_properties.py

```diff
@@ -0,0 +1,345 @@
+#################################################################################
+# The Institute for the Design of Advanced Energy Systems Integrated Platform
+# Framework (IDAES IP) was produced under the DOE Institute for the
+# Design of Advanced Energy Systems (IDAES), and is copyright (c) 2018-2021
+# by the software owners: The Regents of the University of California, through
+# Lawrence Berkeley National Laboratory,  National Technology & Engineering
+# Solutions of Sandia, LLC, Carnegie Mellon University, West Virginia University
+# Research Corporation, et al.  All rights reserved.
+#
+# Please see the files COPYRIGHT.md and LICENSE.md for full copyright and
+# license information.
+#################################################################################
+"""
+Thermophysical property package to be used in conjucntion with ASM1 reactions.
+"""
+
+# Import Pyomo libraries
+import pyomo.environ as pyo
+
+# Import IDAES cores
+from idaes.core import (
+    declare_process_block_class,
+    MaterialFlowBasis,
+    PhysicalParameterBlock,
+    StateBlockData,
+    StateBlock,
+    MaterialBalanceType,
+    EnergyBalanceType,
+    LiquidPhase,
+    Component,
+    Solute,
+    Solvent,
+)
+from idaes.core.util.model_statistics import degrees_of_freedom
+from idaes.core.util.initialization import fix_state_vars, revert_state_vars
+import idaes.logger as idaeslog
+
+# Some more inforation about this module
+__author__ = "Andrew Lee"
+
+
+# Set up logger
+_log = idaeslog.getLogger(__name__)
+
+
+@declare_process_block_class("ASM1ParameterBlock")
+class ASM1ParameterData(PhysicalParameterBlock):
+    """
+    Property Parameter Block Class
+    """
+
+    def build(self):
+        """
+        Callable method for Block construction.
+        """
+        super().build()
+
+        self._state_block_class = ASM1StateBlock
+
+        # Add Phase objects
+        self.Liq = LiquidPhase()
+
+        # Add Component objects
+        self.H2O = Solvent()
+
+        self.inert_soluble = Solute(doc="Soluble inert organic matter")
+        self.substrate = Solute(doc="Readily biodegradable substrate")
+        self.inert_particulate = Solute(doc="Particulate inert organic matter")
+        self.biodegradable = Solute(doc="Slowly biodegradable substrate")
+        self.heterotrophic = Solute(doc="Active heterotrophic biomass")
+        self.autotrophic = Solute(doc="Active autotrophic biomass")
+        self.decay_particulate = Solute(
+            doc="Particulate products arising from biomass decay"
+        )
+        self.oxygen = Solute(doc="Oxygen")
+        self.nitrates = Solute(doc="Nitrate and nitrite nitrogen")
+        self.ammonium = Solute(doc="NH4+ + NH3 nitrogen")
+        self.nitrogen_soluble = Solute(doc="Soluble biodegradable organic nitrogen")
+        self.nitrogen_particulate = Solute(
+            doc="Particulate biodegradable organic nitrogen"
+        )
+
+        self.alkalinity = Component(doc="Alkalinity")
```

Comment:
Trivial suggestion, but might be useful to put the original symbol names in the doc strings for those familiar with the typical ASM formulation. That way, anyone looking behind the scenes can more readily match these components with the naming conventions from ASM1.

Reply:
I'd definitely agree, some of the same components are represented by different symbols in different models, so having the original symbol names would be very helpful